File: test/leetcode/string/remove-all-adjacent-duplicates-ii.test.ts
import { removeDuplicates } from '../../src/string/remove-all-adjacent-duplicates-ii';

describe('remove all adjacent duplicates ii', () => {
  test('remove all adjacent duplicates ii - test case 1', async () => {
    expect(removeDuplicates('abcd', 2)).toBe('abcd');
  });

  test('remove all adjacent duplicates ii - test case 2', async () => {
    expect(removeDuplicates('deeedbbcccbdaa', 3)).toBe('aa');
  });
});

File: test/leetcode/string/remove_all_adjacent_duplicates_ii_test.py
from leetcode.string.remove_all_adjacent_duplicates_ii import Solution


soln = Solution()


def test_case_1():
    assert soln.removeDuplicates("abcd", 2) == "abcd"


def test_case_2():
    assert soln.removeDuplicates("deeedbbcccbdaa", 3) == "aa"
